### DISCUSSION TITLE
Unsigned offset sizes

### DISCUSSION THREAD

**Discussion by JosiahParry** [apache/arrow-rs#5107] [Q&A] [answered]
I'm new to arrow & arrow-rs but had a question about the OffsetSizeTrait. I noticed that it is only implemented for `i32` and `i64`. I find it interesting that it is not implemented for `u32` or `u64` as well. From my naive understanding, offsets would always be positive integers and not negative ones. Out of pure curiosity, what is the reason for not being able to use unsigned ints for an offset?

https://github.com/apache/arrow-rs/blob/df69ef57d055453c399fa925ad315d19211d7ab2/arrow-array/src/array/list_array.rs#L40

**Comment by tustvold** [comment, marked as answer]
The arrow format specification exclusively uses signed offsets - https://arrow.apache.org/docs/format/Columnar.html#list-layout

I believe this is due to some languages, such as Java, historically not having good support for unsigned integers

**Reply by JosiahParry** [reply]
Make enough sense! Thank you :)
